### PR TITLE
[mcp bug bash]: formsg fix

### PR DESCRIPTION
**Pull request by m0nggh**
## What?

Three independent AI Builder / FormSG fixes bundled from the MCP bug bash:

1. **Testing a FormSG trigger always uses mock data.** `execute_step` now passes `testRunMetadata: { preferMock: true }` into `testStep`, so the AI Builder never waits on (or silently reuses) a real form submission when it tests a step inline in chat.
2. **Stale `[MRF]` connection labels no longer block the AI Builder.** Connections verified before #1939 removed the MRF-tagging logic can still carry a `[MRF] ` prefix baked into their stored `screenName`. The AI Builder's connection picker surfaces that stale label, and since the system prompt elsewhere says MRF is unsupported, the model reasonably refuses to use an old "fake MRF" connection that isn't actually MRF.
3. **A visually urgent warning treatment for high-consequence clarification questions.** Adds an `isWarning` flag to `ClarificationQuestion`, driven by a `WARNING: true` line in the `CLARIFICATION_DATA` block — used for the connection-override confirmation (overriding disconnects another pipe's trigger).

## Why?

- (1) AI Builder testing happens inline in chat with no real user action to wait on — there's no "go submit your form now" moment, so mock data is the only sensible default.
- (2) The `[MRF]` tag removal (#1939) fixed new/re-verified connections but never backfilled already-verified ones, since the tag only refreshes on re-verification. Old connections are stuck showing a tag that no longer reflects reality.
- (3) The existing neutral clarification card doesn't communicate that a choice has a real, hard-to-reverse consequence outside the current pipe (e.g. breaking another pipe's connection) — users can click past it without registering the risk.

## What changed?

- `packages/backend/src/services/mcp/execute-step.ts`: always passes `preferMock: true` in `testRunMetadata`. Apps that don't recognize `preferMock` (i.e. everything except FormSG's `newSubmission` trigger) just ignore it via their own zod schema.
- `packages/backend/src/apps/formsg/common/check-live-mrf-status.ts` (new): live-checks FormSG's public schema for a form's *current* MRF status (`responseMode === 'multirespondent' && workflow.length > 0`, matching every other MRF check in the codebase). Returns `null` (not `false`) on a fetch error or missing form, so callers can fail safe.
- `packages/backend/src/services/mcp/list-connections.ts`: `resolveMcpConnectionLabel` detects a stale `[MRF] ` tag on a FormSG connection's label, live-checks the form, and strips the tag display-only (no DB write) only when the check positively confirms the form isn't MRF. Any ambiguity — no `formId`, unparseable `formId`, fetch failure — leaves the label as-is.
- `packages/backend/src/routes/api/chat/parse-clarification-block.ts` / `schema.ts`: adds `isWarning?: boolean` to `ClarificationQuestion`, parsed from an optional `WARNING: true` line directly after `Q:` in a `CLARIFICATION_DATA` block.
- `packages/frontend/src/hooks/useChatStream.ts` / `ChoicePicker.tsx`: mirrors the type on the frontend; renders the warning card (red background/border, warning triangle icon, bold question text) for both the initial question and its review-mode summary.

## Tests

- `execute-step.itest.ts`: asserts `testStep` is called with `testRunMetadata: { preferMock: true }`.
- `list-connections.itest.ts`: new `describe('stale [MRF] label correction (PLU-866)')` block — strips the tag on a confirmed-non-MRF form (including when an env prefix like `[STAGING]` is present), keeps it when the live check confirms MRF or fails, and confirms no live check runs at all for a label with no stale tag or a non-FormSG connection.

## Deploy notes

None — no migrations or config changes. The stale-label fix is read-time only; nothing needs backfilling.

### CODE REVIEW

**Comment by kevinkim-ogp** [comment]
### Code review

Found 1 issue:

1. Forcing `preferMock: true` for every AI Builder FormSG test routes mock-data generation through a stale, unfixed MRF check that will misclassify ordinary (non-workflow) forms as MRF

[execute-step.ts#L34-L41](https://github.com/opengovsg/plumber/blob/010ab0cf9a1f2526563ff5f292a3cbf584028f71/packages/backend/src/services/mcp/execute-step.ts#L34-L41) now unconditionally sets `testRunMetadata: { preferMock: true }`, which forces `shouldUseMockData = true` for every AI Builder test of a FormSG trigger (previously this path was only hit when `hasNoPastSubmission` was true).

That routes execution through [get-mock-data.ts#L205-L233](https://github.com/opengovsg/plumber/blob/010ab0cf9a1f2526563ff5f292a3cbf584028f71/packages/backend/src/apps/formsg/triggers/new-submission/get-mock-data.ts#L205-L233), whose `isMrf` check is `formDetails.form.responseMode === 'multirespondent'` — missing the `workflow.length > 0` refinement that PR #1938 added to the equivalent checks in `get-form-schema.ts` and `new-submission/index.ts`. Since new FormSG forms default to `responseMode: 'multirespondent'` even with no workflow configured (per PR #1939, which this PR's own `check-live-mrf-status.ts` comment references), `get-mock-data.ts` will misclassify these as MRF and incorrectly strip `section`-type field answers from the generated mock data (line 232) — and with `preferMock` now unconditional, this hits on essentially every AI Builder FormSG test, not just an edge case.

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by kevinkim-ogp — APPROVED** [review]
lgtm, tested the changes and they work 👍 
can take a look at claude's comment, think it caught something that we should fix, but might be better to fix directly on `develop-v2` rather than here

File: packages/backend/src/services/mcp/list-connections.ts [hunk 1]
```diff
@@ -62,10 +108,12 @@ export async function listConnectionsService(
...
-    label: connectionLabel(c),
-  }))
+  return Promise.all(
+    connections.map(async (c) => ({
+      id: c.id,
+      appKey: c.key,
+      verified: c.verified,
+      label: await resolveMcpConnectionLabel(c),
+    })),
+  )
```
[the remark this answers is not in the document]
Comment (m0nggh) [reply packages/backend/src/services/mcp/list-connections.ts:130]: !valid ok will just add a concurrency check

File: packages/backend/src/services/mcp/list-connections.ts [hunk 2]
```diff
@@ -22,6 +27,57 @@ export function connectionLabel(c: {
...
+  if (!rawFormId) {
+    return label
+  }
+
+  try {
+    const formId = parseFormIdFromInput(rawFormId)
+    const env = parseFormEnvFromInput(rawFormId)
+    const isCurrentlyMrf = await mrfLiveCheckLimit(() =>
+      checkLiveMrfStatus(formId, env),
+    )
```
[the remark this answers is not in the document]
Comment (m0nggh) [reply packages/backend/src/services/mcp/list-connections.ts:70]: !accepted_risk The core enabler here — `updateConnection` letting a user freely rewrite `formattedData` (including `screenName`/`formId`) with no per-app validation, plus no cap on connections per user — is a pre-existing, broader trust gap in the connection model, not something scoped to this PR. We already capped burst impact per call with `p-limit(5)` (previous finding). Closing the remaining sustained-repeated-call vector properly means fixing that root `updateConnection` validation gap (or adding real rate limiting to `/api/connections`), which we're tracking separately rather than bolting a Redis cache onto AI Builder's MCP services as a one-off (would be the first use of that pattern there). Revisiting once the `updateConnection` hardening lands.
